flocker/node/agents/ebs.py

```diff
@@ -706,8 +707,18 @@ def list_volumes(self):
         """
         Return all volumes that belong to this Flocker cluster.
         """
+        try:
+            ebs_volumes = self.connection.get_all_volumes()
+        except EC2ResponseError as e:
+            # Work around some internal race-condition in EBS by retrying,
+            # since this error makes no sense:
+            if e.code == NOT_FOUND:
+                return self.list_volumes()
```

Comment:
Hm. Well, if we have infinite failures then presumably the agent won't succeed at doing anything anyway, so it doesn't matter?